crates/rattler_package_streaming/src/tokio/async_read.rs

```diff
@@ -33,14 +34,18 @@ pub async fn extract_tar_bz2(
 pub async fn extract_conda(
     reader: impl AsyncRead + Send + 'static,
     destination: &Path,
+    extract_fn: fn(Box<dyn Read>, &Path) -> Result<ExtractResult, ExtractError>,
```

Comment:
I would prefer to keep the API the same as the blocking one, so splitting this up into seperate functions instead of passing in a closure.

Reply:
got it, kept the existing api and pushed this logic a level deeper.